overlord/ifacestate/ifacestate.go

```diff
@@ -67,6 +67,17 @@ func checkConnectConflicts(st *state.State, change *state.Change, plugSnap, slot
 		}
 
 		k := task.Kind()
+		if k == "connect" {
+			var autoConnect bool
+			if err := task.Get("auto", &autoConnect); err != nil && err != state.ErrNoState {
+				return err
+			}
+			// allow only one "connect" task with auto=true (fot autoconnects) at a time
+			if autoConnect {
+				return &state.Retry{After: connectRetryTimeout}
```

Comment:
this look like what we discussed but the code above will turn it into ignore instead of wait. We really need to reduce the ignore case I think to a non-ready symmetric "auto-connect" task, for the other conflicts it should be wait (not ignore). Notice that it means we need to either go over tasks twice, or in one go stop only if we find auto-connect otherwise remember if we saw a different conflict, so that we can decide:

* any not-ready auto-connect for the other side => ignore
* no auto-connect, but other conflict for the other side => wait